stdlib/multiprocessing/heap.pyi

```diff
@@ -0,0 +1,35 @@
+import sys
+from _typeshed import Incomplete
+from collections.abc import Callable
+from mmap import mmap
+from typing_extensions import Protocol, TypeAlias
+
+__all__ = ["BufferWrapper"]
+
+class Arena:
+    size: int
+    buffer: mmap
+    if sys.platform == "win32":
+        name: str
+        def __init__(self, size: int) -> None: ...
+    else:
+        fd: int
+        def __init__(self, size: int, fd: int = ...) -> None: ...
+
+_Block: TypeAlias = tuple[Arena, int, int]
+
+if sys.platform != "win32":
+    class _SupportsDetach(Protocol):
+        def detach(self) -> int: ...
+
+    def reduce_arena(a: Arena) -> tuple[Callable[[int, Incomplete], Arena], tuple[int, Incomplete]]: ...
+    def rebuild_arena(size: int, dupfd: Incomplete) -> Arena: ...
```

Comment:
There's no point defining the protocol if you don't actually use it in the type hints :)

```suggestion
    def reduce_arena(a: Arena) -> tuple[Callable[[int, _SupportsDetach], Arena], tuple[int, Incomplete]]: ...
    def rebuild_arena(size: int, dupfd: _SupportsDetach) -> Arena: ...
```

Reply:
D'oh! Fixing it, sorry about that